Javatrix.java

```diff
@@ -117,5 +115,85 @@ public double get(int i, int j) throws ArrayIndexOutOfBoundsException
     {
         return matrix;
     }
+
+    /**
+     * Get column dimension.
+     *
+     * @return n number of columns
+     */
+    public int getColumnDimension()
+    {
+        return n;
+    }
+
+    /**
+     * Get row dimension.
+     *
+     * @return m number of rows
+     */
+    public int getRowDimension()
+    {
+        return m;
+    }
+
+    /**
+     * Copy the internal two-dimensional array.
+     *
+     * @return matrixCopy copy of internal array
+     */
+    public double[][] getArrayCopy()
+    {
+        double[][] matrixCopy = new double[m][n];
+        for (int i = 0; i < m; i++)
+        {
+            for (int j = 0; j < n; j++)
+            {
+                matrixCopy[i][j] = matrix[i][j];
+            }
+        }
+        return matrixCopy;
+    }
```

Comment:
Looking at the API, it seems like this is supposed to return a Javatrix object with the copied matrix, because the method header says the return type is Matrix.